pkg/functionconfig/types.go

```diff
@@ -488,24 +488,33 @@ const (
 	FunctionStateConfiguringResources             FunctionState = "configuringResources"
 	FunctionStateReady                            FunctionState = "ready"
 	FunctionStateError                            FunctionState = "error"
+	FunctionStateUnhealthyError                   FunctionState = "unhealthyError"
 	FunctionStateScaledToZero                     FunctionState = "scaledToZero"
 	FunctionStateImported                         FunctionState = "imported"
 )
 
-func FunctionStateInSlice(a FunctionState, list []FunctionState) bool {
-	for _, b := range list {
-		if b == a {
+func FunctionStateInSlice(functionState FunctionState, functionStates []FunctionState) bool {
+	for _, state := range functionStates {
+		if functionState == state {
 			return true
 		}
 	}
 	return false
 }
 
+func FunctionStateErrored(functionState FunctionState) bool {
```

Comment:
since there is only one usage for that function + `FunctionStateUnhealthy` is already declared (look up), I will remove this function and check it explicitly (tbh I couldnt come up with a better naming)